src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,194 @@
+            },
+            "Expected UserDataInvalidException for login length less than required minimum");
+  }
+
+  @Test
+  void shouldThrowException_whenPasswordIsNull() {
+    User user = new User();
+    user.setLogin("validLogin");
+    user.setPassword(null);
+    user.setAge(21);
+    assertThrows(UserDataInvalidException.class, () -> {
+              registrationService.register(user);
+            },
+            "Expected UserDataInvalidException when password is null");
+  }
+
+  @Test
+  void shouldThrowException_whenPasswordIsEmpty() {
+    User user = new User();
+    user.setLogin("validLogin");
+    user.setPassword("");
+    user.setAge(21);
+    assertThrows(UserDataInvalidException.class, () -> {
+              registrationService.register(user);
+            },
+            "Expected UserDataInvalidException when password is empty");
+  }
+
+  @Test
+  void shouldThrowException_whenPasswordIsTooShort() {
+    User user = new User();
+    user.setLogin("validLogin");
+    user.setPassword("123");
+    user.setAge(21);
+    assertThrows(UserDataInvalidException.class, () -> {
+              registrationService.register(user);
+            },
+            "Expected UserDataInvalidException when password length is less than required minimum");
+  }
+
+  @Test
+  void shouldThrowException_whenAgeIsNull() {
+    User user = new User();
+    user.setLogin("validLogin");
+    user.setPassword("validPassword123");
+    user.setAge(null);
+    assertThrows(UserDataInvalidException.class, () -> {
+              registrationService.register(user);
+            },
+            "Expected UserDataInvalidException when age is null");
+  }
+
+  @Test
+  void shouldThrowException_whenAgeIsNegative() {
+    User user = new User();
+    user.setLogin("validLogin");
+    user.setPassword("validPassword123");
+    user.setAge(-25);
+    assertThrows(UserDataInvalidException.class, () -> {
+              registrationService.register(user);
+            },
+            "Expected UserDataInvalidException when age is negative");
+  }
+
+  @Test
+  void shouldThrowException_whenAgeIsBelowMinimum() {
+    User user = new User();
+    user.setLogin("validLogin");
+    user.setPassword("validPassword123");
+    user.setAge(17);
+    assertThrows(UserDataInvalidException.class, () -> {
+              registrationService.register(user);
+            },
+            "Expected UserDataInvalidException when age is below minimum required");
+  }
+
+  @Test
+  void shouldThrowException_whenLoginAlreadyExists() {
+    User user = new User();
+    user.setLogin("validLogin");
+    user.setPassword("validPassword123");
+    user.setAge(50);
+    Storage.people.add(user);
+    assertThrows(UserDataInvalidException.class, () -> {
+              registrationService.register(user);
+            },
+            "Expected UserDataInvalidException when login already exists in storage");
+  }
+
+  @Test
+  void shouldRegisterValidUser() {
```

Comment:
The test method names such as `shouldThrowException_whenUserIsNull` should follow the convention `<methodUnderTest>_<state>_<expectedBehavior>`. For example, `register_nullUser_notOk`.

Reply:
changed